libcloud/compute/drivers/outscale.py

```diff
@@ -0,0 +1,1109 @@
+
+        :param      private_ips: One or more private IP addresses of the VM.
+        :type       private_ips: ``list``
+
+        :param      security_group_ids: One or more IDs of security group
+        for the VMs.
+        :type       security_group_ids: ``list``
+
+        :param      security_groups: One or more names of security groups
+        for the VMs.
+        :type       security_groups: ``list``
+
+        :param      subnet_id: The ID of the Subnet in which you want to
+        create the VM.
+        :type       subnet_id: ``str``
+
+        :return: the created instance
+        :rtype: ``dict``
+        """
+        data = {
+            "DryRun": dry_run,
+            "BootOnCreation": boot_on_creation,
+            "BsuOptimized": bsu_optimized,
+            "ImageId": image_id
+        }
+        if block_device_mapping is not None:
+            data.update({"BlockDeviceMappings": block_device_mapping})
+        if client_token is not None:
+            data.update({"ClientToken": client_token})
+        if deletion_protection is not None:
+            data.update({"DeletionProtection": deletion_protection})
+        if keypair_name is not None:
+            data.update({"KeypairName": keypair_name})
+        if max_vms_count is not None:
+            data.update({"MaxVmsCount": max_vms_count})
+        if min_vms_count is not None:
+            data.update({"MinVmsCount": min_vms_count})
+        if nics is not None:
+            data.update({"Nics": nics})
+        if performance is not None:
+            data.update({"Performance": performance})
+        if placement is not None:
+            data.update({"Placement": placement})
+        if private_ips is not None:
+            data.update({"PrivateIps": private_ips})
+        if security_group_ids is not None:
+            data.update({"SecurityGroupIds": security_group_ids})
+        if security_groups is not None:
+            data.update({"SecurityGroups": security_groups})
+        if subnet_id is not None:
+            data.update({"SubnetId": subnet_id})
+        action = "CreateVms"
+        data = json.dumps(data)
+        signer = OSCRequestSignerAlgorithmV4(access_key=self.key,
+                                             access_secret=self.secret,
+                                             version=self.version,
+                                             connection=self.connection)
+        headers = signer.get_request_headers(action=action,
+                                             data=data,
+                                             service_name=self.service_name,
+                                             region=self.region)
+        endpoint = self._get_outscale_endpoint(self.region,
+                                               self.version,
+                                               action)
+        return requests.post(endpoint, data=data, headers=headers)
+
+    def reboot_node(self, node_ids):
```

Comment:
To comply with the libcloud standard API, this method needs to take a single ``node`` argument which must be ``Node`` class instance.

If you also want to support rebooting multiple notes, you should add a new extension method. For example: ``ex_reboot_nodes(self, node_ids)``.